languages/scala/reference/exercise-concepts/hamming.md

```diff
@@ -0,0 +1,32 @@
+# Concepts of hamming
+
+[Example implementation](https://github.com/exercism/scala/blob/master/exercises/hamming/example.scala)
+
+## General
+
+- object or case class: used as class wrapper for exercies
+- functions: used as main entry point for the exercise
+- function arguments: input strands passed as arguments
+- return values: return a value from a function
+- conditionals using if: conditionally execute logic using an `if` statement
+- strings: `string` passed as the two input parameters
+- character operations: `!=`
+- integer: `Int` is used as return result within `Option`
+- Option: `Option` is used as return result
+- math operations: `==`
+- zip: combine two strings into a sequence of tuples
+- count: count number of lambda calls that return true
```

Comment:
Similar concept to before, where I think this is an implementation of a more general concept like "collections - folding" or "collections-reducing" or something like that?

Reply:
You said `reduce` in my PRs i think

Reply:
Right. Whoops. I don't mind us not using the exact same terminology, we'll fix this when compiling the final list.